proto/regen/ecocredit/v1alpha1/tx.proto

```diff
@@ -217,4 +226,52 @@ message MsgCancel {
 }
 
 // MsgCancelResponse is the Msg/Cancel response type.
-message MsgCancelResponse {}
\ No newline at end of file
+message MsgCancelResponse {}
+
+// MsgUpdateClassAdmin is the Msg/UpdateClassAdmin request type.
+message MsgUpdateClassAdmin {
```

Comment:
we could also perhaps just squeeze all these new messages into 1 MsgUpdateClass -- and just have each field (admin, metadata, issuers) as optional? not sure which approach is more ideal here

Reply:
+1 for using a single message type.

Reply:
i looked into this and wasn't sure what the best way to implement it is - specifically with the issuers type ( []string ). I'm unsure if there's a good way in proto to tell if a field has been explicitly set. otherwise, it appears a bit troublesome to figure out intent via the default values

Reply:
so looking at this again, I could just make each update field its own message i.e.:

```proto
message MsgUpdateCreditClass {
   string admin = 1;
   string class_id = 2;
   UpdateMetadata metadata = 3;
  // ...
}

message UpdateMetadata {
   bytes metadata = 1;
}
```

that way each update field will be a pointer and we can check intent via nil. Does this seem like a sane path forward?

Reply:
I think the way you have it now with separate update messages would be best. In most cases, I would imagine the admin would only be changing one field at a time. It's also done this way in the group module.